src/app/explore/explore.component.ts

```diff
@@ -152,41 +342,116 @@ export class ExploreComponent implements OnDestroy, OnInit {
+          query: this.normaliseQueryForRecentChildrenComparison(query),
+          offset: this.offset,
+          children: this.individualCampaigns,
+          time: Date.now(), // ms
+        };
+
+        this.sessionStorage.set(this.recentChildrenKey, recentChildrenData);
+      }
     }, () => {
+      this.filterError = true; // Error, should only be thrown if the callout SF API returns an error
       this.loading = false;
     });
   }
 
+  private normaliseQueryForRecentChildrenComparison(query: SearchQuery): string {
+    delete query.offset;
+
+    return JSON.stringify(query); // We don't want to get into object key / true equality comparisons, so just JSON it.
+  }
+
   private run() {
     this.searched = this.searchService.nonDefaultsActive;
 
     this.offset = 0;
-    const query = this.campaignService.buildQuery(this.searchService.selected, 0);
-    this.campaigns = [];
+    const query = this.campaignService.buildQuery(this.searchService.selected, 0, this.metaCampaign?.id, this.campaignSlug, this.fundSlug);
+    this.individualCampaigns = [];
     this.loading = true;
 
     this.campaignService.search(query as SearchQuery).subscribe(campaignSummaries => {
-      this.campaigns = campaignSummaries; // Success
+      this.individualCampaigns = campaignSummaries; // Success
       this.loading = false;
     }, () => {
         this.loading = false;
       },
     );
+
+    if (!isPlatformBrowser(this.platformId)) { // Server renders don't need the scroll restoration help
+      // this.doCampaignSearch(query as SearchQuery, true); // Clear existing children, though there _should_ be none on server
+      return;
+    }
+
+    const recentChildrenData = this.sessionStorage.get(this.recentChildrenKey);
+    // Only an exact query match should reinstate the same child campaigns on load.
+    if (
+      recentChildrenData &&
+      recentChildrenData.time > (Date.now() - (60000 * this.recentChildrenMaxMinutes)) &&
+      recentChildrenData.query === this.normaliseQueryForRecentChildrenComparison(query as SearchQuery)
+    ) {
+      this.individualCampaigns = recentChildrenData.children;
+      // We need to separately reinstate the offset, while excluding it from the normalised query params
+      // we use for equality comparison, so that moreMightExist() and therefore scrolling to load more
+      // campaigns still works after we reinstate the existing children.
+      this.offset = recentChildrenData.offset;
+
+      // Auto scrolling without a significant extra wait only works when
+      // the child campaigns were quickly loaded from local state from
+      // a recent page view.
+      if (this.navigationService.getLastScrollY() >= this.smallestSignificantScrollPx) {
+        this.shouldAutoScroll = true;
+      }
+
+      this.loading = false;
+
+      return;
+    }
+
+    // Else need to load children newly in browser.
+    // this.doCampaignSearch(query as SearchQuery, true); // Clear existing children
```

Comment:
This line's `MetaCampaignComponent` was uncommented I believe. Do we need to keep it but make it conditional now?

Reply:
Thanks, fixed in pairing with @dorota-joanna yesterday.